docs/cli_hacking_guide.adoc

```diff
@@ -0,0 +1,293 @@
+= OpenShift 3 Command Line Interface (CLI)
```

Comment:
Agreed, header changed. I kept the introduction for people that reach this doc first, but added links to cli.md.